### PR TITLE
Fixed median algorithm bug

### LINKED ISSUE #436: Condition is not fully correct in quicksort [Python]
Consider this case `arr[k]>arr[i]>arr[j]` in which arr[i] should be median but we get arr[k] as median instead.

- [ ] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by j-haj**
Fixes #436 

Modified the algorithm slightly to first find the median value, then find the index corresponding to that value. The basic idea is to sum the values, then remove the max and min values, leaving the median as the result.

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

File: quick_sort/quick_sort.py [hunk 1]
```diff
@@ -7,17 +7,13 @@ def median(a, i, j, k):
...
-        else:
-            return k
-
+    ai, aj, ak = a[i], a[j], a[k]
+    med_val = ai + aj + ak - max(ai, aj, ak) - min(ai, aj, ak)
+    if ai == med_val:
+        return i
+    elif aj == med_val:
+        return j
+    return k
```
Comment (singhpratyush) [inline quick_sort/quick_sort.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp4__xq2to/quick_sort/quick_sort.py
+++ b/tmp/tmp4__xq2to/quick_sort/quick_sort.py
@@ -14,6 +14,7 @@
     elif aj == med_val:
         return j
     return k
+
 
 def partition(array, l, r):
     """
```

Comment (j-haj) [reply quick_sort/quick_sort.py]: Ah yes - my apologies. My pep8 vim plugin did not catch this for some reason.... updating the PR now!

**Comment by j-haj** [comment]
@singhpratyush I added an extra newline and updated the commit

**Pull request merged**